packages/medusa/src/commands/migrate.ts

```diff
@@ -1,19 +1,50 @@
 import Logger from "../loaders/logger"
-import { migrateMedusaApp, revertMedusaApp } from "../loaders/medusa-app"
+import { runMedusaAppMigrations } from "../loaders/medusa-app"
 import { initializeContainer } from "../loaders"
 import { ContainerRegistrationKeys, MedusaError } from "@medusajs/utils"
 import { getResolvedPlugins } from "../loaders/helpers/resolve-plugins"
 import { resolvePluginsLinks } from "../loaders/helpers/resolve-plugins-links"
 
 const TERMINAL_SIZE = process.stdout.columns
 
+type Action = "run" | "revert" | "generate" | "show"
+
+function validateInputArgs({
+  action,
+  modules,
+}: {
+  action: Action
+  modules: string[]
+}) {
+  const actionsRequiringModules = ["revert", "generate"]
+
+  if (modules.length && !actionsRequiringModules.includes(action)) {
+    Logger.error(
+      `<modules> cannot be specified with the "${action}" action. Please remove the <modules> argument and try again.`
+    )
+    process.exit(1)
+  }
+
+  if (!modules.length && actionsRequiringModules.includes(action)) {
+    Logger.error(
+      "Please provide the modules for which you want to revert migrations"
+    )
+    Logger.error(`For example: "npx medusa migration revert <moduleName>"`)
```

Comment:
suggestion(non-blocking): Should we change `<moduleName>` to an actual module name from our core suite to make it really obvious what they should do?

Reply:
We display a list of modules after this example.

My fear with saying `npx medusa migration revert user` is that some might think they always have to run the same exact command.

Reply:
Yeah, I see. Makes sense. Fine to keep as is 👍